fluent-syntax/src/parser.js

```diff
@@ -15,24 +28,36 @@ export function parse(source) {
   const entries = [];
 
   while (ps.current()) {
-    const entryStartPos = ps.getIndex();
+    const spanFrom = ps.getIndex();
 
     try {
       const entry = getEntry(ps);
-      if (entryStartPos === 0 && entry.type === 'Comment') {
+      if (spanFrom === 0 && entry.type === 'Comment') {
         comment = entry;
       } else {
+        entry.addSpan(spanFrom, ps.getIndex());
         entries.push(entry);
       }
-    } catch (e) {
-      entries.push(getJunkEntry(ps, source, entryStartPos));
-      errors.push(e);
+    } catch (err) {
+      ps.skipToNextEntryStart();
+
+      const spanTo = ps.getIndex();
+      const slice = source.substring(spanFrom, spanTo);
+      const junk = new AST.Junk(slice);
+      junk.addSpan(spanFrom, spanTo);
+      entries.push(junk);
+
+      errors.push(Object.assign(err, {
+        lineNumber: lineOffset(source, err.pos) + 1,
+        columnNumber: columnOffset(source, err.pos) + 1,
```

Comment:
In compare-locales, I use 1-based indices for both, because I want to give user-digestible positions.

I think toolchains are inconsistent enough that we don't need to care about matching what they expect.